cupy_backends/cupy_nccl.h

```diff
@@ -166,4 +172,44 @@ ncclResult_t ncclRecv(void* recvbuff, size_t count, ncclDataType_t datatype,
 }
 #endif
 
+#if (NCCL_VERSION_CODE < NCCL_VERSION(2, 18, 0))
+// mock `ncclConfig_t` with our own struct
+typedef struct {
+    // expose more fields if necessary
+    int splitShare;
+} _ncclConfig_t;
+
+static inline _ncclConfig_t ncclConfigInitializer() {
+    return _ncclConfig_t{0};
+}
+
+ncclResult_t _ncclCommSplit(ncclComm_t comm, int color, int key, ncclComm_t *newcomm,
+                            _ncclConfig_t *config) {
+    return ncclSuccess;
+}
+
+ncclResult_t _ncclCommInitRankConfig(ncclComm_t *comm, int nranks, ncclUniqueId commId,
+                                     int rank, _ncclConfig_t *config) {
+    return ncclSuccess;
+}
+
+#else // #if (NCCL_VERSION_CODE >= NCCL_VERSION(2, 18, 0))
+// reuse the original struct
+typedef ncclConfig_t _ncclConfig_t;
+
+static inline _ncclConfig_t ncclConfigInitializer() {
+    return (_ncclConfig_t)NCCL_CONFIG_INITIALIZER;
+}
```

Comment:
Could probably just do:
```
#ifndef NCCL_CONFIG_INITIALIZER
#define NCCL_CONFIG_INITIALIZER _ncclConfig_t{0}
#endif
```
I suspect (the `#ifdef` since IIRC the config addition doesn't align with 2.18).  But maybe not worth re-running CI over.